src/primitives.jl

```diff
@@ -161,11 +167,18 @@ unwrapping `Nullable` entries. A copy is always returned, even when
 `X` does not contain any null values.
 """
 function dropnull{T}(X::AbstractVector{T})                  # -> AbstractVector
-    if !(Nullable <: T) && !(T <: Nullable)
+    u = isa(T, Union)
```

Comment:
Can you move this logic to a `isnullable` function? That will avoid code duplication and will also allow testing directly for corner cases like unions and `Any`.